crates/compiler/gen_dev/src/generic64/aarch64.rs

```diff
@@ -2302,6 +2708,320 @@ fn udiv_reg64_reg64_reg64(
+            rd: dst,
+            rn: src,
+        });
+
+    buf.extend(inst.bytes());
+}
+
+/// `FDIV Sd/Dd, Sn/Dn, Sm/Dm` -> Divide Sn/Dn by Sm/Dm and place the result into Sd/Dd.
+#[inline(always)]
+fn fdiv_freg_freg_freg(
+    buf: &mut Vec<'_, u8>,
+    ftype: FloatWidth,
+    dst: AArch64FloatReg,
+    src1: AArch64FloatReg,
+    src2: AArch64FloatReg,
+) {
+    let inst =
+        FloatingPointDataProcessingTwoSource::new(FloatingPointDataProcessingTwoSourceParams {
+            opcode: 0b0001,
+            ptype: ftype,
+            rd: dst,
+            rn: src1,
+            rm: src2,
+        });
+
+    buf.extend(inst.bytes());
+}
+
+/// `FMOV Sd/Dd, Sn/Dn` -> Move Sn/Dn to Sd/Dd.
+#[inline(always)]
+fn fmov_freg_freg(
+    buf: &mut Vec<'_, u8>,
+    ftype: FloatWidth,
+    dst: AArch64FloatReg,
+    src: AArch64FloatReg,
+) {
+    let inst =
+        FloatingPointDataProcessingOneSource::new(FloatingPointDataProcessingOneSourceParams {
+            opcode: 0b000000,
+            ptype: ftype,
+            rd: dst,
+            rn: src,
+        });
+
+    buf.extend(inst.bytes());
+}
+
+/// Encode a 32-bit float into an 8-bit immediate for FMOV.
+/// See Table C2-1 in the ARM manual for a table of every float that can be encoded in 8 bits.
+/// If the float cannot be encoded, return None.
+/// This operation is the inverse of VFPExpandImm in the ARM manual.
+#[inline(always)]
+fn encode_f32_to_imm8(imm: f32) -> Option<u8> {
+    let n = 32;
+    let e = 8; // number of exponent bits in a 32-bit float
+    let f = n - e - 1; // 23: number of fraction bits in a 32-bit float
+
+    let bits = imm.to_bits();
+
+    let sign = (bits >> (n - 1)) & 1; // bits<31>
+    let exp = (bits >> f) & ((1 << e) - 1); // bits<30:23>
+    let frac = bits & ((1 << f) - 1); // bits<22:0>
+
+    let exp_first = (exp >> (e - 1)) & 1; // exp<7>
+    let exp_middle = (exp >> 2) & ((1 << (e - 3)) - 1); // exp<6:2>
+    let exp_last = exp & 0b11; // exp<1:0>
+    if exp_first == 0 && exp_middle != ((1 << (e - 3)) - 1) {
+        // If exp_first is 0, exp_middle must be all 1s.
+        return None;
+    }
+    if exp_first == 1 && exp_middle != 0 {
+        // If exp_first is 1, exp_middle must be all 0s.
+        return None;
+    }
+
+    let frac_begin = frac >> (f - 4); // frac<22:19>
+    let frac_end = frac & ((1 << (f - 4)) - 1); // frac<18:0>
+    if frac_end != 0 {
+        // frac_end must be all 0s.
+        return None;
+    }
+
+    // The sign is the same.
+    let ret_sign = sign << 7;
+    // The first bit of the exponent is inverted.
+    let ret_exp_first = (exp_first ^ 1) << 6;
+    // The rest of the exponent is the same as the last 2 bits of the original exponent.
+    let ret_exp_last = exp_last << 4;
+    // The fraction is the same as the first 4 bits of the original fraction.
+    let ret_frac = frac_begin;
+
+    Some(
+        (ret_sign | ret_exp_first | ret_exp_last | ret_frac)
+            .try_into()
+            .unwrap(),
```

Comment:
This should always be safe. I think we shouldn't need the try and unwrap style conversion.

```suggestion
        (ret_sign | ret_exp_first | ret_exp_last | ret_frac) as u8
```